2-Draft-Accepted/RFCNNN-HelpSystem-As-A-Module.md

```diff
@@ -0,0 +1,50 @@
+---
+RFC: RFC<four digit unique incrementing number assigned by Committee, this shall be left blank by the author>
+Author: Aditya Patwardhan
+Status: Draft
+SupercededBy: <link to another RFC>
+Version: 0.1
+Area: Help System
+Comments Due: 08/15/2017
+Plan to implement: Yes
+---
+
+# Help System as a standalone PowerShell Module  
+
+This RFC proposes to remove the Help System from System.Management.Automation.dll and make it a standalone PowerShell module.
+
+## Motivation
+
+Currently, help system is part of System.Management.Automation.dll. 
+Help system is used predominantly in interactive sessions and hence not required for automation scenarios.
+Since, it is part of System.Management.Automation.dll it is always loaded with PowerShell.
+Help system is sizeable piece of code, which increases the runtime footprint of the process.
+This RFC proposes to make the help system a PowerShell module, so it is loaded only when used.
+
+## Specification
+
+All the help system code resides under [help folder](https://github.com/PowerShell/PowerShell/tree/master/src/System.Management.Automation/help).
+This code will be moved to a separate assembly and be published as a module.
+
+All the public APIs from the help system will be moved to the new namespace 'Microsoft.PowerShell.HelpSystem'.
+This will be breaking change for code taking dependency of these public APIs.
+
+| Public API | Breaking change impact assessment
+| ------------ | --------
+| HelpCategoryInvalidException | Exception class, breaking change impact is low.
+| GetHelpCommand | Sealed class, breaking change impact is low.
+| GetHelpCodeMethods | Used by formatting and output, breaking change impact is medium.
+| HelpNotFoundException | Exception class, breaking change impact is low.
+| SaveHelpCommand | Sealed class, breaking change impact is low.
+| UpdatableHelpCommandBase | Used by UpdateHelpCommand, breaking change impact is medium.
+| UpdateHelpCommand | Sealed class, breaking change impact is low.
+
+[CLR Type Forwarding](https://docs.microsoft.com/en-us/dotnet/framework/app-domains/type-forwarding-in-the-common-language-runtime) will be considered to lessen the impact of the breaking change.
```

Comment:
I think we should state this as main solution.

Reply:
Actually, this might not be feasible as type forwarding only works if the namespace is same.

Reply:
We could place mandatory types in mandatory assembly (maybe leave in SMA) and any extentions put in optional assemblies (and maybe another namespace).
HelpSystem is based on help providers so we could consider to make the help providers as optional and loadable.